syntax = "proto3";

package test.integration.filters;

import "validate/validate.proto";

// Configuration for inspector listener filter
message TestInspectorFilterConfig {
}

// Configuration for TCP listener filter test
message TestTcpListenerFilterConfig {
  uint32 drain_bytes = 1 [(validate.rules).uint32 = {gte: 2}];
}
// Configuration for UDP listener filter test
message TestUdpListenerFilterConfig {
}

// Configuration for QUIC listener filter test
message TestQuicListenerFilterConfig {
  string added_value = 1 [(validate.rules).string = {min_len: 1}];
  bool allow_server_migration = 2;
  bool allow_client_migration = 3;
}
